src/main/ruby/truffleruby/core/gc.rb

```diff
@@ -76,10 +76,55 @@ def garbage_collect
   end
 
   def self.stat(option = nil)
+    time, count, minor_count, major_count, unknown_count, commited, used, memory_pool_names, memory_pool_info = Primitive.gc_stat()
+
+    # Initialize stat for statistics that come from memory pools, and populate it with some final stats
     stat = {
-      count: GC.count,
-      time: GC.time,
+      :count => count,
+      :time => time,
+      :minor_gc_count => minor_count,
+      :major_gc_count => major_count,
+      :unknown_count => unknown_count, # if nonzero, major or minor count needs to be updated for this GC case
+      :committed => 0.0,
+      :init => 0.0,
+      :max => 0.0,
+      :used => 0.0,
+      :peak_committed => 0.0,
+      :peak_init => 0.0,
+      :peak_max => 0.0,
+      :peak_used => 0.0,
+      :last_committed => 0.0,
+      :last_init => 0.0,
+      :last_max => 0.0,
+      :last_used => 0.0,
+      :heap_available_slots => commited, # should be the same as the calculated commited
+      :heap_live_slots => used, # should be the same as the calculated used
+      :heap_free_slots => commited - used,
     }
+
+    (0...memory_pool_names.length).each do |i|
+      # Populate memory pool specific stats
+      stat[Truffle::Interop.to_string(memory_pool_names[i])] = {
```

Comment:
`Truffle::Interop.to_string` should be removed once https://github.com/oracle/truffleruby/pull/2280/files#r585822976 is done (same below)